packages/alfa-rules/src/sia-er62/rule.ts

```diff
@@ -383,26 +397,68 @@ function hasNonLinkText(device: Device): Predicate<Element> {
 namespace Distinguishable {
   export function isDistinguishable(
     container: Element,
+    target: Element,
     device: Device,
     context: Context = Context.empty()
-  ): Predicate<Element> {
-    return or(
+  ): Predicate<Element>[] {
+    let predicates: Array<
```

Comment:
We should keep the same order as they used to be (more or less).
There are two group of distinguishing factors: those that need to be different from the container, and those whose mere presence is enough. The new order is mixing these two groups.
We can still `.sort` the array immediately upon building, since the code doesn't care about the order.